core/lib/dal/src/connection/mod.rs

```diff
@@ -218,34 +254,50 @@ impl ConnectionPool {
     }
 
     /// A version of `access_storage` that would also expose the duration of the connection
-    /// acquisition tagged to the `requester` name.
+    /// acquisition tagged to the `requester` name. It also tracks the caller location for the purposes
+    /// of logging (e.g., long-living connections) and debugging (when used with a test connection pool).
     ///
     /// WARN: This method should not be used if it will result in too many time series (e.g.
     /// from witness generators or provers), otherwise Prometheus won't be able to handle it.
-    pub async fn access_storage_tagged(
+    #[track_caller] // In order to use it, we have to de-sugar `async fn`
```

Comment:
Theoretically, tracking caller should only introduce some executable size overhead due to additional code locations included. This seems acceptable given that we don't have that many call locations (my IDE tells there are 96).